pkgs/applications/misc/waybar/default.nix

```diff
@@ -21,6 +22,11 @@
 
     nativeBuildInputs = [
       meson ninja pkgconfig scdoc wrapGAppsHook cmake
+    ] ++ stdenv.lib.optional withMediaPlayer playerctl;
+
+    propagatedBuildInputs = stdenv.lib.optionals withMediaPlayer [
+      gobject-introspection
```

Comment:
this should be `native`, and you'll need to set `strictDeps = false;` for it's setup hook to work properly.